src/apiTest/kotlin/com/ampnet/auditornode/script/api/classes/WebSocketInputApiTest.kt

```diff
@@ -0,0 +1,233 @@
+            @Language("JavaScript") val scriptSource = jsAssertions + """
+                function audit() {
+                    let fields = [
+                        {
+                            "type": "boolean",
+                            "name": "booleanField",
+                            "description": "test1"
+                        },
+                        {
+                            "type": "number",
+                            "name": "numberField",
+                            "description": "test2"
+                        },
+                        {
+                            "type": "string",
+                            "name": "stringField",
+                            "description": "test3"
+                        }
+                    ];
+                    let values = Input.readFields(fields, "test");
+
+                    assertEquals("values.get(\"booleanField\")", true, values.get("booleanField"));
+                    assertEquals("values.get(\"numberField\")", 42, values.get("numberField"));
+                    assertEquals("values.get(\"stringField\")", "string field", values.get("stringField"));
+                    return AuditResult.of(true);
+                }
+            """.trimIndent()
+
+            val result = client.toBlocking().retrieve(
+                HttpRequest.POST("/script/store", scriptSource).apply {
+                    contentType(MediaType.TEXT_PLAIN_TYPE)
+                }
+            )
+
+            val responseRegex = """^\{"id":"($UUID_REGEX)"}$""".toRegex()
+            val matchResult = responseRegex.find(result)
+                ?: fail("Response does not match regular expression: $responseRegex")
+
+            storedScriptId = matchResult.groups[1]?.value
+            assertThat(storedScriptId).isNotNull()
+        }
+
+        verify("correct values are returned") {
+            val client = webSocketClient.connect(WebSocketTestClient::class.java, "/script/interactive/$storedScriptId")
+                .blockingFirst()
+            client.assertNextMessage(ConnectedInfoMessage)
+            client.assertNextMessage(ExecutingInfoMessage)
+
+            val fields = listOf(
+                InputField(
+                    type = InputType.BOOLEAN,
+                    name = "booleanField",
+                    description = "test1"
+                ),
+                InputField(
+                    type = InputType.NUMBER,
+                    name = "numberField",
+                    description = "test2"
+                ),
+                InputField(
+                    type = InputType.STRING,
+                    name = "stringField",
+                    description = "test3"
+                )
+            )
+
+            client.assertNextMessage(ReadFieldsCommand("test", fields))
+            client.send("true")
+            client.send("42")
+            client.send("string field")
```

Comment:
Eh, I would leave this as is in tests, especially because there is this annoying thing in IntelliJ - if you use string interpolation then referenced variables are also marked as JavaScript which then displays annoying syntax error highlight which is not there.